manifests/user.pp

```diff
@@ -13,63 +13,94 @@
 #   `1` - CALLBACK
 #
 #   Controls the rights of the user to be created.
+# @param enable
+#   Should this user be enabled?
 # @param user_id
-#   The user id of the user to be created. Should be unique from existing users. On
-#   SuperMicro IPMI, user id 2 is reserved for the ADMIN user.
+#   The user id of the user to be created. Should be unique from existing users.
+#   On SuperMicro IPMI, user id 2 is reserved for the 'ADMIN' username.
+#   On ASUS IPMI, user id 2 is reserved for the 'admin' username.
 #
 define ipmi::user (
-  Variant[Sensitive[String[1]], String[1]] $password,
   String $user     = 'root',
   Integer $priv    = 4,
+  Boolean $enable  = true,
   Integer $user_id = 3,
+  Optional[Variant[Sensitive[String[1]], String[1]]] $password = undef,
 ) {
-  require ipmi
+  require ipmi::install
 
-  case $priv {
-    1: { $privilege = 'CALLBACK' }
-    2: { $privilege = 'USER' }
-    3: { $privilege = 'OPERATOR' }
-    4: { $privilege = 'ADMINISTRATOR' }
-    default: { fail('invalid privilege level specified') }
-  }
+  if $enable {
+    if empty($password) {
```

Comment:
Is this check necessary? I believe the type annotation should prevent `$password` from being an empty string.

Reply:
With `enable==false` the `password` isn't required, the type annotation changed to permit omitting the password.  But there doesn't seem to be a way in the type defs to say "if X==true then Y must be Z"